config/api.yml

```diff
@@ -56,6 +56,20 @@
       :get:
       - :name: read
         :identifier: embedded_configuration_script_payload_view
+  :configuration_script_sources:
+    :description: Configuration Script Source
+    :options:
+    - :collection
+    :verbs: *g
+    :klass: ConfigurationScriptSource
+    :collection_actions:
+      :get:
+      - :name: read
+        :identifier: embedded_configuration_script_source_view
```

Comment:
Although, I have to agree with @bdunne about the need for differentiating anything at the API level.  A user should just get all of the configuration_script_sources they can see, regardless of what providers they are tied to.  If they want ones from a particular provider, they would go through a subcollection.

I understand needing to filter the collection, but how do we do it with respect to something like VMs and Instances (i.e. Infra vs Cloud)